test/users/signup.test.js

```diff
@@ -0,0 +1,70 @@
+const supertest = require('supertest');
+const { factory } = require('factory-girl');
+const app = require('../../app');
+const { UserData, UserErrors } = require('../factory/user_factory');
+
+const request = supertest(app);
+
+describe('/users POST', () => {
+  test('Create user must be success', async () => {
+    const { newUser: user, responseWhenUserIsCreated: userResponse } = UserData;
+    const response = await request.post('/users').send(user);
+    expect(response.statusCode).toBe(201);
+    expect(response.body).toMatchObject(userResponse);
+  });
+  test('Create user when email exists must fail', async () => {
+    const { newUser: user } = UserData;
+    const userCreated = await factory.create('UserWithHash');
+    const response = await request.post('/users').send({ ...user, email: userCreated.email });
+    expect(response.statusCode).toBe(400);
+    expect(response.body).toMatchObject(UserErrors.emailExists);
```

Comment:
I think UserErrors should be matched with the real errors, not with a factored object

Reply:
userErrors are real, but I replaced the strings with constants.🙂